From 95bdbf9b1558b5b8a37d62f0bbc23be3a2f6a33e Mon Sep 17 00:00:00 2001
From: Kai Uwe Broulik <kde@privat.broulik.de>
Date: Tue, 2 Jan 2024 13:12:01 +0100
Subject: [PATCH] QWaylandEglWindow: Pass window surface format to
 q_glFormatFromConfig

Otherwise it disregards the window's swap interval preference.

Fixes: QTBUG-120477
Pick-to: 6.7 6.6 6.5
Change-Id: Icc6b192f9f317a80acdf4b588ee3f1c26f2ddf0a
---
 .../client/wayland-egl/qwaylandeglwindow.cpp                    | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/hardwareintegration/client/wayland-egl/qwaylandeglwindow.cpp b/src/hardwareintegration/client/wayland-egl/qwaylandeglwindow.cpp
index 4d7d4787..f4ed8cd6 100644
--- a/src/hardwareintegration/client/wayland-egl/qwaylandeglwindow.cpp
+++ b/src/hardwareintegration/client/wayland-egl/qwaylandeglwindow.cpp
@@ -110,7 +110,7 @@ void QWaylandEglWindow::updateSurface(bool create)
             if (mDisplay->supportsWindowDecoration())
                 fmt.setAlphaBufferSize(8);
             EGLConfig eglConfig = q_configFromGLFormat(m_clientBufferIntegration->eglDisplay(), fmt);
-            m_format = q_glFormatFromConfig(m_clientBufferIntegration->eglDisplay(), eglConfig);
+            m_format = q_glFormatFromConfig(m_clientBufferIntegration->eglDisplay(), eglConfig, fmt);
             m_eglSurface = eglCreateWindowSurface(m_clientBufferIntegration->eglDisplay(), eglConfig, eglw, 0);
             if (Q_UNLIKELY(m_eglSurface == EGL_NO_SURFACE))
                 qCWarning(lcQpaWayland, "Could not create EGL surface (EGL error 0x%x)\n", eglGetError());
-- 
2.42.0

